core/src/main/scala/org/apache/spark/util/collection/ExternalSorter.scala

```diff
@@ -736,11 +734,16 @@ private[spark] class ExternalSorter[K, V, C](
       val writeStartTime = System.nanoTime
       util.Utils.tryWithSafeFinally {
         for (i <- 0 until numPartitions) {
-          val in = new FileInputStream(partitionWriters(i).fileSegment().file)
-          util.Utils.tryWithSafeFinally {
-            lengths(i) = org.apache.spark.util.Utils.copyStream(in, out, false, transferToEnabled)
-          } {
-            in.close()
+          val file = partitionWriters(i).fileSegment().file
+          if (!file.exists()) {
+            lengths(i) = 0
+          } else {
+            val in = new FileInputStream(file)
+            util.Utils.tryWithSafeFinally {
```

Comment:
Shouldn't we avoid using partial package names like this? Just import spark.util.Utils.